ProjFS.Linux/PrjFSLib.Linux.Managed/VirtualizationInstance.cs

```diff
@@ -291,7 +291,7 @@ private static string GetProcCmdline(int pid)
                     return parts.Length > 0 ? parts[0] : string.Empty;
                 }
             }
-            catch (IOException ex) when (ex is FileNotFoundException || ex is DirectoryNotFoundException)
+            catch
```

Comment:
This has me feeling a bit on edge. What other kinds of exceptions are there?

Perhaps we should continue to `return string.Empty;` in all cases, but log the error if the exception isn't an expected `IOException` per the current logic?

Reply:
Unfortunately I don't think there's a logging infra this deep in the stack ... the GVFS provider has its `Tracer` framework, but here we don't have access to that, AFAIK.

The errors I've seen tend be due to races and occur when recovering from a sleep state, but all that's in the `gvfs_mount_process....log` is `Unhandled Exception`, I believe.  I've been trying to reproduce them for a few days, but they are something of a heisenbug.

In the end, though, I'm not sure it matters.  The code here has to be robust against all exceptions, regardless of their cause, and the retrieval of the process's command line is -- as far as I can tell -- purely informational.  It's used in the higher-level code in its [logging](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs#L356) in `OnGetFileStream()` callbacks, and in its [telemetry](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Virtualization/FileSystemCallbacks.cs#L461) [statistics](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Virtualization/FileSystemCallbacks.cs#L297) collected by [`CreatePlaceholders()`](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs#L644) in its `OnEnumerateDirectory()` callbacks.

Your question did prompt me to notice that there's no need to call this on the non-projection notifications, because the result is unused!  So we can actually pull out a big chunk of code here!  See 97189a1 for a nice cleanup -- thank you for the prompt!!

Reply:
> The code here has to be robust against all exceptions, regardless of their cause, and the retrieval of the process's command line is — as far as I can tell — purely informational. It's used in the higher-level code only in its [logging](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs#L356) in `OnGetFileStream()` callbacks, and in the [telemetry](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Virtualization/FileSystemCallbacks.cs#L461) [statistics](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Virtualization/FileSystemCallbacks.cs#L297) collected by [`CreatePlaceholders()`](https://github.com/microsoft/VFSForGit/blob/ceaaa10d19aa587e5eb2cac63dfce74d70f43749/GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs#L644) in its `OnEnumerateDirectory()` callbacks

:+1::+1: Makes sense!

> Your question did prompt me to notice that there's no need to call this on the non-projection notifications, because the result is unused! So we can actually pull out a big chunk of code here! See [d902623](https://github.com/microsoft/VFSForGit/commit/d9026232d0cdd705f508a4780ed2914f00a0c3bf) for a nice cleanup -- thank you for the prompt!!

Super good result!

<img width="123" alt="image" src="https://user-images.githubusercontent.com/1915/58297258-6cbbb080-7e1a-11e9-89d8-51a3dac5c734.png">